### ISSUE TITLE
Worker unable to join HA cluster (token is not valid: 401 Unauthorized)

### ISSUE THREAD

**Issue by japsu** [rancher/k3s#2470]
**Environmental Info:**
K3s Version: k3s version v1.19.3+k3s1 (974ad30b)

Node(s) CPU architecture, OS, and Version: Linux qb3 5.4.0-1031-azure #32-Ubuntu SMP Tue Oct 6 09:47:33 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration: High Availability with Embedded etcd. 3 masters, trying to add 1 worker

**Describe the bug:**

I have been successfully able to create a HA k3s cluster following [this guide](https://rancher.com/docs/k3s/latest/en/installation/ha-embedded/). However, when I try to add a fourth node that is not a master but just a worker, its log (`journalctl -u k3s-agent`) shows 

```
Nov 02 21:31:45 qb4 k3s[8896]: time="2020-11-02T21:31:45.110928545Z" level=error msg="token is not valid: https://127.0.0.1:32991/apis: 401 Unauthorized"
```

Server logs (qb1) show:
```
Nov 02 21:31:04 qb1 k3s[849]: time="2020-11-02T21:31:04.931485651Z" level=info msg="Cluster-Http-Server 2020/11/02 21:31:04 http: TLS handshake error from 10.9.1.5:55492: remote error: tls: bad certificate
```

Token triple checked to be the same on all three servers (some characters redacted):

```
(venv3-ansible) japsu@Korok:~/Hobby/ansible-tracon$ ansible -u root -a 'cat /var/lib/rancher/k3s/server/node-token' k3s_servers,k3s_initial_server
qb3 | CHANGED | rc=0 >>
K101c77a12f46ee926027...aa87a365::server:bd35...4c93e
qb2 | CHANGED | rc=0 >>
K101c77a12f46ee926027...aa87a365::server:bd35...4c93e
qb1 | CHANGED | rc=0 >>
K101c77a12f46ee926027...aa87a365::server:bd35...4c93e
```

**Steps To Reproduce:**

- Installed K3s: 

Exact installation command for initial server qb1:

```yaml
- name: install k3s-server
  shell: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION='{{ k3s_version }}' INSTALL_K3S_EXEC='server --disable traefik --node-ip {{ k3s_node_ip_address }} --cluster-init' sh -
  args:
    creates: /etc/systemd/system/k3s.service
```

For servers qb2 and qb3

```yaml
- name: install k3s-server
  shell: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION='{{ k3s_version }}' K3S_TOKEN='{{ k3s_token }}' INSTALL_K3S_EXEC='server --disable traefik --node-ip {{ k3s_node_ip_address }} --server https://{{ k3s_server_ip_address }}:6443' sh -
  args:
    creates: /etc/systemd/system/k3s.service
```

For worker qb4

```yaml
- name: install k3s-agent
  shell: curl -sfL https://get.k3s.io | K3S_URL=https://{{ k3s_server_ip_address }}:6443 K3S_TOKEN={{ k3s_token }} sh -
  args:
    creates: /etc/systemd/system/k3s-agent.service
```

**Expected behavior:**

`qb4` node provisioned and joins the cluster, shows up on `kubectl get node`

**Actual behavior:**

Node doesn't join the cluster, log messages above

**Additional context / logs:**

**Comment by brandond** [comment]
Can you try again with the latest release? Many issues with managed etcd were fixed between v1.19.3+k3s1 and v1.19.3+k3s2.

**Comment by japsu** [comment]
Thanks for the answer! I had missed that there's a new K3s release as they're the same Kubernetes version. I'll upgrade the cluster and try again. Also if the problem persists I'll try to reproduce it in a fresh Vagrant environment.

**Comment by japsu** [comment]
Fixed in 1.19.3+k3s2, thanks!

**Comment by hackingwhangjuncomputer** [comment]
I updated the version but still getting the error
